init.lua

```diff
@@ -30,6 +30,20 @@ local function destroy(handle)
 end
 ffi.gc(cudnn.handle, destroy)
 
+cudnn.typemap = {
+   ['torch.CudaHalfTensor']   = 'CUDNN_DATA_HALF',
+   ['torch.CudaTensor']       = 'CUDNN_DATA_FLOAT',
+   ['torch.CudaDoubleTensor'] = 'CUDNN_DATA_DOUBLE',
+}
+
+-- TODO: determine if device supports true half and use true half on it
```

Comment:
Actually, after I have rebuilt with proper architectures, DilatedConvolution does seem to pass, but I am having those errors in cunn test:
Completed 704 asserts in 148 tests with 0 failures and 2 errors
--------------------------------------------------------------------------------
VolumetricMaxUnpooling_backward_batch
 Function call failed
/usr/share/lua/5.1/nn/THNN.lua:109: bad argument #2 to 'v' (input image smaller than kernel size at /tmp/buildd/torch7-0.9.99/extra/nn/lib/THNN/generic/VolumetricMaxPooling.c:146)
stack traceback:
        [C]: in function 'v'
        /usr/share/lua/5.1/nn/THNN.lua:109: in function 'VolumetricMaxPooling_updateOutput'
        /usr/share/lua/5.1/nn/VolumetricMaxPooling.lua:45: in function 'forward'
        /usr/share/lua/5.1/cunn/test.lua:4455: in function 'v'
        /usr/share/lua/5.1/cunn/test.lua:5576: in function </usr/share/lua/5.1/cunn/test.lua:5574>
        [C]: in function 'xpcall'
        /usr/share/lua/5.1/torch/Tester.lua:477: in function '_pcall'
        /usr/share/lua/5.1/torch/Tester.lua:436: in function '_run'
        /usr/share/lua/5.1/torch/Tester.lua:355: in function 'run'
        /usr/share/lua/5.1/cunn/test.lua:5596: in function 'testcuda'
        [string "nn.testcuda()"]:1: in main chunk
        [C]: in function 'pcall'
        /usr/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
        [C]: at 0x00406670

--------------------------------------------------------------------------------
VolumetricMaxUnpooling_forward_batch
 Function call failed
/usr/share/lua/5.1/nn/THNN.lua:109: bad argument #2 to 'v' (input image smaller than kernel size at /tmp/buildd/torch7-0.9.99/extra/nn/lib/THNN/generic/VolumetricMaxPooling.c:146)
stack traceback:
        [C]: in function 'v'
        /usr/share/lua/5.1/nn/THNN.lua:109: in function 'VolumetricMaxPooling_updateOutput'
        /usr/share/lua/5.1/nn/VolumetricMaxPooling.lua:45: in function 'forward'
        /usr/share/lua/5.1/cunn/test.lua:4403: in function 'v'
        /usr/share/lua/5.1/cunn/test.lua:5576: in function </usr/share/lua/5.1/cunn/test.lua:5574>
        [C]: in function 'xpcall'
        /usr/share/lua/5.1/torch/Tester.lua:477: in function '_pcall'
        /usr/share/lua/5.1/torch/Tester.lua:436: in function '_run'
        /usr/share/lua/5.1/torch/Tester.lua:355: in function 'run'
        /usr/share/lua/5.1/cunn/test.lua:5596: in function 'testcuda'
        [string "nn.testcuda()"]:1: in main chunk
        [C]: in function 'pcall'
        /usr/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
        [C]: at 0x00406670

--------------------------------------------------------------------------------
/usr/share/lua/5.1/torch/Tester.lua:363: An error was found while running tests!